src/python/pose_format/bin/directory.py

```diff
@@ -144,15 +198,26 @@ def main():
 
     pose_with_no_errors_count = 0
 
-    for vid_path in tqdm(videos_with_missing_pose_files):
-        try:
-            pose_path = get_corresponding_pose_path(video_path=vid_path, keep_video_suffixes=args.keep_video_suffixes)
-            if pose_path.is_file():
-                print(f"Skipping {vid_path}, corresponding .pose file already created.")
-                continue
-            pose_video(vid_path, pose_path, args.format, additional_config)
-            pose_with_no_errors_count += 1
-        except ValueError as e:
-            print(f"ValueError on {vid_path}")
-            logging.exception(e)
+    if args.num_workers == 1:
+        print('Process sequentially ...')
+
+        for vid_path in tqdm(videos_with_missing_pose_files):
+            success = process_video(vid_path, args.keep_video_suffixes, args.format, additional_config)
+            if success:
+                pose_with_no_errors_count += 1
+    else:
+        print(f'Multiprocessing with {args.num_workers} workers on {len(os.sched_getaffinity(0))} available CPUs ...')
+
+        with Pool(args.num_workers) as pool:
+            params = [[
+                vid_path, 
+                args.keep_video_suffixes, 
+                args.format,
+                additional_config,
+             ] for vid_path in videos_with_missing_pose_files]
+
+            for success in tqdm(pool.istarmap(process_video, params), total=len(params)):
```

Comment:
if you want both `multiprocessing` and `tqdm`, I think it might be better to use:
```py
from tqdm.contrib.concurrent import process_map
```

Reply:
Last time I took a swing at this with MultiProcessing, that's how I did it, specifically https://stackoverflow.com/a/63834834 so that the progress bar updates more intuitively. I used it in my split_folder.py:
```
    def move_files(subfolder_idx, file_group):
        # Format subfolder name with zero-padding
        subfolder_path = src_path / f'split_{str(subfolder_idx).zfill(max_digits)}'
        subfolder_path.mkdir(exist_ok=True)
        print(f"subfolder: {subfolder_path}, file group type: {type(file_group)}")

        for file_list in file_group:
            for file in file_list:
                shutil.move(str(file), subfolder_path / file.name)

    def distribute_file_groups():
        for i in range(0, total_file_groups, groups_per_split):
            yield grouped_files[i:i+groups_per_split]

    with ThreadPoolExecutor(max_workers=workers) as executor:
        for idx, file_group in enumerate(distribute_file_groups()):
            executor.submit(move_files, idx + 1, file_group)

    print(f"Split {total_file_groups} file groups into {idx + 1} subfolders.")
```

Reply:
Some more options discussed here: https://stackoverflow.com/questions/41920124/multiprocessing-use-tqdm-to-display-a-progress-bar